presto-password-authenticators/src/test/java/io/prestosql/plugin/password/salesforce/TestSalesforceBasicAuthenticator.java

```diff
@@ -0,0 +1,269 @@
+    }
+
+    @Test
+    public void createAuthenticatedPrincipalAllOrgs()
+    {
+        org = "all";  // As if from salesforce.org property.
+        username = "user@salesforce.com";
+        password = "passtoken";
+
+        config = new SalesforceConfig()
+                .setOrgs(org);
+
+        String xmlResponse = String.format(successResponse, "some18CharOrgId", username);
+
+        testHttpClient = new TestingHttpClient((request -> mockResponse(HttpStatus.OK, MediaType.ANY_TEXT_TYPE, xmlResponse)));
+        SalesforceBasicAuthenticator authenticator = new SalesforceBasicAuthenticator(config, testHttpClient);
+
+        Principal principal = authenticator.createAuthenticatedPrincipal(username, password);
+        assertEquals(principal.getName(), username, "Test allowing all orgs.");
+    }
+
+    @Test
+    public void createAuthenticatedPrincipalFewOrgs()
+    {
+        org = "my18CharOrgId,your18CharOrgId, his18CharOrgId ,her18CharOrgId";  // As if from salesforce.org property.
+        username = "user@salesforce.com";
+        password = "passtoken";
+
+        config = new SalesforceConfig()
+                .setOrgs(org);
+
+        String xmlResponse = String.format(successResponse, "my18CharOrgId", username);
+
+        testHttpClient = new TestingHttpClient((request -> mockResponse(HttpStatus.OK, MediaType.ANY_TEXT_TYPE, xmlResponse)));
+        SalesforceBasicAuthenticator authenticator = new SalesforceBasicAuthenticator(config, testHttpClient);
+
+        Principal principal = authenticator.createAuthenticatedPrincipal(username, password);
+        assertEquals(principal.getName(), username, "Test allowing a few orgs.");
+    }
+
+    /*
+     * Real tests that use Salesforce credentials and actually attempt to login.
+     * These should be disabled for automated builds and test runs.
+     *
+     * In order to run these, the following environment variables need to be set.
+     *
+     *   - SALESFORCE_TEST_ORG (this is the 18 character organization id)
+     *   - SALESFORCE_TEST_USERNAME
+     *   - SALESFORCE_TEST_PASSWORD (this must be password and security token concatenation)
+     *   - SALESFORCE_TEST_FORREAL must be TRUE
+     */
+
+    // Test a real login.
+    @Test(description = "Test principal name for real, yo!")
+    void createAuthenticatedPrincipal_realSuccess()
+    {
+        // Skip this test if SALESFORCE_TEST_FORREAL is not set to TRUE.
+        if (!forReal) {
+            throw new SkipException("Skipping real tests.");
+        }
+
+        org = System.getenv("SALESFORCE_TEST_ORG");
+        if (org == null || org.length() == 0) {
+            fail("Must set SALESFORCE_TEST_ORG environment variable.");
+        }
+        username = System.getenv("SALESFORCE_TEST_USERNAME");
+        password = System.getenv("SALESFORCE_TEST_PASSWORD");
+        if (username == null || username.length() == 0 || password == null || password.length() == 0) {
```

Comment:
this form `username.length() == 0` should be replaced with `username.isEmpty()`. Your IDE should be making this suggestion for you automatically.

Reply:
Didn't get any hint on it from IntelliJ, but will research perhaps why.  I've replaced all these checks with `emptyToNull` as suggested.  These checks are out of old habit.  I should form new habits.